unity-renderer/Assets/DCLPlugins/ECS7/Systems/SceneBoundsCheckerSystem/ECSSceneBoundsCheckerSystem.cs

```diff
@@ -0,0 +1,249 @@
+                sbcComponentData.entity.isInsideSceneBoundaries = sbcComponentData.scene.IsInsideSceneBoundaries(globalBoundsMaxPoint + worldOffset, globalBoundsMaxPoint.y)
+                                                                  && sbcComponentData.scene.IsInsideSceneBoundaries(globalBoundsMinPoint + worldOffset);
+
+                // 3. If merged bounds is detected as outside bounds we need a final check on submeshes (for L-Shaped subdivided meshes)
+                if (!sbcComponentData.entity.isInsideSceneBoundaries)
+                    sbcComponentData.entity.isInsideSceneBoundaries = AreSubMeshesAndCollidersInsideBounds(sbcComponentData);
+            }
+            else
+            {
+                sbcComponentData.entity.isInsideSceneBoundaries = false;
+            }
+
+            SetInsideBoundsStateForMeshComponents(sbcComponentData);
+        }
+
+        private bool AreSubMeshesAndCollidersInsideBounds(ECSComponentData<InternalSceneBoundsCheck> sbcComponentData)
+        {
+            var renderers = sbcComponentData.model.renderers;
+            var physicsColliders = sbcComponentData.model.physicsColliders;
+            var pointerColliders = sbcComponentData.model.pointerColliders;
+            int renderersCount = renderers?.Count ?? 0;
+            int collidersCount = physicsColliders?.Count ?? 0 + pointerColliders?.Count ?? 0;
+
+            // For entities with 1 mesh/collider the already-checked merged bounds already represent its bounds
+            // So we avoid all these unneeded submesh checks for those
+            if (renderersCount + collidersCount <= 1) return sbcComponentData.entity.isInsideSceneBoundaries;
+
+            if (renderers != null)
+            {
+                for (int i = 0; i < renderersCount; i++)
+                {
+                    if (!sbcComponentData.scene.IsInsideSceneBoundaries(renderers[i].bounds))
+                        return false;
+                }
+            }
+
+            if (physicsColliders != null)
+            {
+                int physicsCollidersCount = physicsColliders.Count;
+                for (int i = 0; i < physicsCollidersCount; i++)
+                {
+                    if (!sbcComponentData.scene.IsInsideSceneBoundaries(physicsColliders[i].bounds))
+                        return false;
+                }
+            }
+
+            if (pointerColliders != null)
+            {
+                int pointerCollidersCount = pointerColliders.Count;
+                for (int i = 0; i < pointerCollidersCount; i++)
+                {
+                    if (!sbcComponentData.scene.IsInsideSceneBoundaries(pointerColliders[i].bounds))
+                        return false;
+                }
+            }
+
+            return true;
+        }
+
+        private bool AnyRendererOrColliderComponentRemoved(ECSComponentData<InternalSceneBoundsCheck> sbcComponentData)
```

Comment:
we should review about modifying `sbcComponentData.model` inside this function. 
the function's name is misleading I wouldn't expect a function called `AnyRendererOrColliderComponentRemoved` to modify stuff I expect to return a boolean given an input